app/src/main/java/com/aptoide/uploader/apps/view/MyStoreFragment.java

```diff
@@ -80,13 +106,38 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     storeNameText = null;
     recyclerView.setAdapter(null);
     recyclerView = null;
+    selectionObservable.dispose();
     logoutConfirmation.dismiss();
     logoutConfirmation = null;
     logoutItem = null;
     toolbar = null;
     super.onDestroyView();
   }
 
+  @Override public void resetSelectionState(){
+
+    if(adapter.getSelectedCount()!=0){
+      toggleSubmitButton(false);
+      selectionObservable.dispose();
+      setUpSelectionListener();
+    }
+  }
+
+  private void setUpSelectionListener(){
+    selectionObservable = adapter.toggleSelection()
+        .flatMap(status -> handleTitleChange(status))
```

Comment:
we are not doing anything with the status inside handleTitleChange, shouldn't this be a doOnNext() ? we want to change the title without changing the chain object.

Reply:
Fixed